### PR TITLE
Chore: convert `VariableOptions` to use emotion instead of scss

### LINKED ISSUE #64134: Convert `./public/app/features/variables/pickers/shared/VariableOptions.tsx` to use emotion

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- converts `VariableOptions` to use emotion instead of scss
- this also fixes a minor styling issue with the dropdown padding (see https://github.com/grafana/grafana/pull/56690#issuecomment-1452955055)

| before | after |
| --- | --- |
| image | image |

**Why do we need this feature?**

- more consistent styling

**Who is this feature for?**

- grafana devs

**Which issue(s) does this PR fix?**:

Fixes #64134 

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/variables/pickers/shared/VariableOptions.tsx [hunk 1]
```diff
@@ -96,35 +110,98 @@ class VariableOptions extends PureComponent<Props> {
...
+      background: `url(${checkboxImageUrl}) 0px -36px no-repeat`,
+    }),
+    variableOptionIconSelected: css({
+      background: `url(${checkboxImageUrl}) 0px -18px no-repeat`,
+    }),
+    variableValueDropdown: css({
+      backgroundColor: theme.colors.background.primary,
+      border: `1px solid ${theme.colors.border.weak}`,
+      borderRadius: theme.shape.borderRadius(2),
+      boxShadow: theme.shadows.z2,
```
Comment (ashharrison90) [inline public/app/features/variables/pickers/shared/VariableOptions.tsx:184]: looking at the before, maybe this should be `z3`? seems a bit harsh though 🤔

**Review by eledobleefe — APPROVED** [review]
LGTM 👍

File: public/sass/components/_submenu.scss [hunk 2]
```diff
@@ -30,114 +21,7 @@
   }
 }
 
-.variable-value-link {
```
Comment (nmarrs) [inline public/sass/components/_submenu.scss:33]: How confident are we that this scss is not being relied on elsewhere in the codebase? 🤔

**Review by nmarrs — APPROVED** [review]
LGTM provided we can verify removed scss is not relied on elsewhere 👍

File: public/sass/components/_submenu.scss [hunk 3]
```diff
@@ -8,15 +8,6 @@
   padding: 0 0 $space-sm 0;
 }
 
-.annotation-segment {
```
Comment (axelavargas) [inline public/sass/components/_submenu.scss:11]: ❓ I couldn't find the relationship between this class and the changes on the PR, is this just a deprecating class you are removing?

**Review by axelavargas — APPROVED** [review]
Awesome work, Mr @ashharrison90 😎, codewise looks great! and I also gave it a spin, it works as expected.

File: public/sass/components/_submenu.scss (hunk 2 above)
Comment (ashharrison90) [reply public/sass/components/_submenu.scss:33]: pretty confident. [i searched the entire grafana org for `variable-value-link`](https://github.com/search?q=org%3Agrafana+variable-value-link&type=code) and the only hit in code is from an [abandoned app plugin](https://github.com/grafana/kentik-app/blob/12ea573ccd5ea5886e220cd9f32abd0b7c485377/src/styles/kentik-main.scss#L612). 

there's no real way of knowing whether people are using this as a selector externally to override css. they might be. but i don't think we can account for that tbh, else we'll never be able to change a classname ever 😅

File: public/sass/components/_submenu.scss (hunk 3 above)
Comment (ashharrison90) [reply public/sass/components/_submenu.scss:11]: yeah similar to above, i searched the entire codebase for `annotation-segment` and nowhere is using the class name.

**Pull request merged**